src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -103,22 +103,44 @@ public byte[] GetAddressBytes()
             return (byte[])_address.Clone();
         }
 
-        public static PhysicalAddress Parse(string address)
+        public static PhysicalAddress Parse(string address) => address != null ? Parse(address.AsSpan()) : None;
+
+        public static PhysicalAddress Parse(ReadOnlySpan<char> address)
         {
-            int validSegmentLength;
-            char? delimiter = null;
-            byte[] buffer;
+            if (!TryParseInternal(address, out PhysicalAddress value))
+                ThrowBadAddressException(address);
 
+            return value;
+
+            static void ThrowBadAddressException(ReadOnlySpan<char> address) =>
+                throw new FormatException(SR.Format(SR.net_bad_mac_address, new string(address)));
+        }
+
+        public static bool TryParse(string address, out PhysicalAddress value)
+        {
             if (address == null)
             {
-                return None;
+                value = None;
+                return true;
             }
 
+            return TryParseInternal(address, out value);
+        }
+
+        public static bool TryParse(ReadOnlySpan<char> address, out PhysicalAddress value) => TryParseInternal(address, out value);
+
+        private static bool TryParseInternal(ReadOnlySpan<char> address, out PhysicalAddress value)
```

Comment:
Why do we need a separate TryParseInternal, rather than just having this be the implementation for `TryParse(ReadOnlySpan<char> address, out PhysicalAddress value)` and having the other methods call that public API?

Reply:
It's just a pattern I'm used to where we extract some common logic required by several public methods into a private method. It's kind of a separation between the public API and the implementation inside the same type. If you have some strong objections I can move the logic to the public TryParse.

Reply:
It's just unnecessary code and overhead for zero benefit.

Reply:
Removed